apps/prairielearn/src/pages/instructorStudents/components/StudentsCard.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  type ColumnDef,
+  type ColumnFiltersState,
+  type ColumnPinningState,
+  type ColumnSizingState,
+  type VisibilityState as ColumnVisibilityState,
+  type SortingState,
+  createColumnHelper,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from '@tanstack/react-table';
+import { parseAsString, useQueryState } from 'nuqs';
+import { useMemo, useState } from 'preact/compat';
+
+import { type StudentRow, parseAsSortingState } from '../instructorStudents.shared.js';
+
+import { ColumnManager } from './ColumnManager.js';
+import { DownloadButton } from './DownloadButton.js';
+import { StudentsTable } from './StudentsTable.js';
+
+const columnHelper = createColumnHelper<StudentRow>();
+
+// This default must be declared outside the component to ensure referential
+// stability across renders, as `[] !== []` in JavaScript.
+const DEFAULT_SORT: SortingState = [];
+
+export function StudentsCard({ students }: { students: StudentRow[] }) {
+  const [globalFilter, setGlobalFilter] = useQueryState('search', parseAsString.withDefault(''));
+  const [sorting, setSorting] = useQueryState<SortingState>(
+    'sort',
+    parseAsSortingState.withDefault(DEFAULT_SORT),
+  );
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+  const [columnVisibility, setColumnVisibility] = useState<ColumnVisibilityState>({});
+  const [columnPinning, setColumnPinning] = useState<ColumnPinningState>({
+    left: [],
+    right: [],
+  });
+
+  const columns = useMemo<ColumnDef<StudentRow, any>[]>(
+    () => [
+      columnHelper.accessor('uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('created_at', {
+        header: 'Enrolled At',
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
```

Comment:
This should include the timezone, and should also be formatted with the configured timezone for the course instance. We typically reach for `formatDate` from `@prairielearn/formatter` for this.